packages/synapse-sdk/src/telemetry/types.ts

```diff
@@ -0,0 +1,147 @@
+export type Environment = 'test' | 'development' | 'staging' | 'production'
+
+export interface TelemetryConfig {
+  /**
+   * Whether to enable telemetry.
+   *
+   * You can also control this via the environment variable `SYNAPSE_TELEMETRY_DISABLED` or the global variable `SYNAPSE_TELEMETRY_DISABLED`.
+   *
+   * This value will also be false if `NODE_ENV === 'test'`.
+   *
+   * @default true
+   */
+  enabled?: boolean
+  environment?: Environment // optional: deployment environment
```

Comment:
do we need this?  I'm wondering if is overkill right now and we maybe wait until we identify a need?